File: PROGRAMMERS/예산.py
# d = [1,3,2,5,4]
# budget = 9
d = [2,2,3,3]
budget = 10


def solution(d, budget):
    answer = -1
    d.sort()
    for i in range(len(d)):
        tmp = d[i]
        cnt = 1
        for j in range(i+1, len(d)):
            tmp += d[j]
            cnt += 1
            if tmp > budget:
                tmp -= d[i]
                cnt -= 1
        if cnt > answer:
            answer = cnt
    return answer

#
# def solution(d, budget):
#     answer = 0
#     d.sort()
#     tmp = 0
#     cnt = 0
#     for i in range(len(d)):
#         tmp += d[i]
#         cnt += 1
#         if tmp > budget:
#             tmp -= d[i]
#             cnt -= 1
#     return answer

print(solution(d, budget))

File: PROGRAMMERS/JAVA/src/PG/Lv_1/예산.java
package PG.Lv_1;


import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class 예산 {
    public static int solution(int[] d, int budget) {
        int answer = 0;
        Arrays.sort(d);
        List<Integer> sum = new ArrayList<>();
        for (int i = 0; i < d.length; i++) {
            sum.add(d[i]);
            if (sum.stream().mapToInt(Integer::intValue).sum() > budget) {
                sum.remove(sum.size()-1);
                break;
            }
        }
        answer = sum.size();
        return answer;
    }

    public static void main(String[] args) {
        int[] d = new int[]{1, 3, 2, 5, 4};
//        int[] d = new int[]{2,2,3,3};
        int result = solution(d, 9);
        System.out.println(result);
    }
}
